valpas-web/src/utils/types.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * A type guard function to to filter null, false and undefined
+ * @param x Param
+ * @returns True, if x is not false, null or undefined
+ */
+export function filterFalsy<T>(
+  x: T | false | undefined
+): x is Exclude<T, null | false | undefined> {
```

Comment:
Huomasin omassa toteutuksessa, että tuo palautustyyppi voi olla sittenkin `x is NonNullable<T> & NonFalsy<T>`.

`NonFalsy<T>` on luotu, jotta TypeScript-kääntäjä huomaa `false`-arvon käytön oikein.

```typescript
type NonFalsy<T> = T extends false | "" ? never : T

export function filterFalsy<T>(
  x: T | "" | false | undefined | null
): x is NonNullable<T> & NonFalsy<T> {
  return x !== undefined && x !== null && x !== false && (typeof x === "string" && x !== "")
}
```